core-processor/src/executor.rs

```diff
@@ -250,11 +250,27 @@ pub fn execute_wasm<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environ
         }
     };
 
+    let is_initial = program.get_allocations().is_empty();
+
+    let exports = program.code().exports();
+    if !exports.contains(&dispatch.kind()) {
+        let allocations = match is_initial {
+            true => Some((0..static_pages.0).map(WasmPageNumber).collect()),
+            false => None,
+        };
```

Comment:
simplify with if clause?

Reply:
Frankly speaking I intentionally made it this way since it looks more concise.

Reply:
> Frankly speaking I intentionally made it this way since it looks more concise.

How do you feel about [this](https://doc.rust-lang.org/std/primitive.bool.html#method.then) stuff?